controller/resource/retryresource/crud_resource_wrapper.go

```diff
@@ -41,6 +44,34 @@ func newCRUDResourceWrapper(config Config) (*crudResourceWrapper, error) {
 		}
 
 		underlyingCRUD, ok := underlying.(*controller.CRUDResource)
+		if !ok {
+			// TODO this is pretty magical. Needs separate testing and sharing between wrapping resources.
+
+			// Check if *controller.CRUDResource is embbeded type.
+			// And if it is extract embbeded CRUDResource.
+
+			// Create compiler checked name of the CRUDResource.
+			crudResourceTypeName := fmt.Sprintf("%T", controller.CRUDResource{})
+			split := strings.Split(crudResourceTypeName, ".")
+			crudResourceTypeName = split[len(split)-1]
+
+			// Find a field with "CRUDResource".
+			val := reflect.ValueOf(underlying)
+			val = reflect.Indirect(val)
+			crudResourceField := val.FieldByName(crudResourceTypeName)
+
+			// If the the field with name "CRUDResource" is a non
+			// zero reflec.Value and it's of of type
+			// *controller.CRUDResource then this underlying
+			// resource embeds *controller.CRUDResource. In that
+			// case it is remembered and ok is set to true.
+			zeroValue := reflect.Value{}
+			if crudResourceField != zeroValue && crudResourceField.Type() == reflect.TypeOf(&controller.CRUDResource{}) {
+				i := crudResourceField.Interface()
+				underlyingCRUD = i.(*controller.CRUDResource)
+				ok = true
+			}
+		}
```

Comment:
After uncommenting tests I discovered we need also to handle the case when *CRUDResource is embedded. This requires reflection, as we need to check if the struct contains field `CRUDResource` of type `*CRUDResource` and instrument it if so.

Reply:
This doesn't affect our current workflows, because we don't embed `CRUDResource`. We tend to embed `Controller` only.

Reply:
How to get it straight?

Reply:
`CRUDResource` is not an interface. It's a struct thus pointer or value. `Resource` is an interface.

Checking if something is of type `*CRUDResource` is easy. But there is also a case when the type embeds `*CRUDResource`. Then it also implements the interface so this code.

Reply:
Yeah I don't get that at all.